monitoring.go

```diff
@@ -237,20 +237,20 @@ func (sc *snowflakeConn) rowsForRunningQuery(
 	resp, err := sc.getQueryResultResp(ctx, resultPath)
 	if err != nil {
 		logger.WithContext(ctx).Errorf("error: %v", err)
-		if resp != nil {
```

Comment:
sc.getQueryResultResp() will only ever return a resp or an error, never both. Changing the flow to either return the error or check the response.